lms/djangoapps/course_goals/api.py

```diff
@@ -0,0 +1,76 @@
+"""
+Course Goals Python API
+"""
+from enum import Enum
+from opaque_keys.edx.keys import CourseKey
+from django.utils.translation import ugettext as _
+from openedx.core.djangolib.markup import Text
+
+from .models import CourseGoal
+
+
+def add_course_goal(user, course_id, goal_key):
+    """
+    Add a new course goal for the provided user and course.
+
+    Arguments:
+        user: The user that is setting the goal
+        course_id (string): The id for the course the goal refers to
+        goal_key (string): The goal key that maps to one of the
+            enumerated goal keys from CourseGoalOption.
+
+    """
+    # Create and save a new course goal
+    course_key = CourseKey.from_string(str(course_id))
+    new_goal = CourseGoal(user=user, course_key=course_key, goal_key=goal_key)
+    new_goal.save()
+
+
+def get_course_goal(user, course_key):
+    """
+    Given a user and a course_key, return their course goal.
+
+    If a course goal does not exist, returns None.
+    """
+    course_goals = CourseGoal.objects.filter(user=user, course_key=course_key)
+    return course_goals[0] if course_goals else None
+
+
+def remove_course_goal(user, course_key):
+    """
+    Given a user and a course_key, remove the course goal.
+    """
+    course_goal = get_course_goal(user, course_key)
+    if course_goal:
+        course_goal.delete()
+
+
+class CourseGoalOption(Enum):
+    """
+    Types of goals that a user can select.
+
+    These options are set to a string goal key so that they can be
+    referenced elsewhere in the code when necessary.
+    """
+    CERTIFY = 'certify'
```

Comment:
I don't think it makes sense to have this along with the Choices.  It seems like we should trade one for the other.  Sorry I didn't look at the Choice code sooner.